cmd/util/hook/hooks.go

```diff
@@ -68,7 +68,9 @@ func AfterParentPostRunHook(hook runHookE) runHookE {
 // should have applied.
 var ClientPreRunHooks runHookE = Chain(
 	Adapt(ApplyPorcelainLogLevel),
-	Adapt(StartUpdateCheck),
+	// TODO(forrest) [fixme/are-you-fucking-kidding-me]: need to un-wang this from here
+	// gut check idea is to perform this in a single place at the root level
+	// Adapt(StartUpdateCheck),
```

Comment:
This whole thing is really gross, especially now that we have moved away from a global config structure. In order to run the update checker we need an API client so we can ask the server what it's version is and then notify the user that the server they are talking to is out of date. I don't think we should notify users their server is out of date - I beleive the utility of this serve should be to tell users their _binary_ is out of date. Further, doing all this in a pre-run hook is really complicated since it requires a client , TLS config, and api token to connect to the server and authenticate.

We still run an update checker that notifies users when they run `serve` but with this commented out we won't notify users when they run client commands. 

I think a the follow on change here ought to be:
1. Dial get.bacalhau.org to get the latest version
2. if the version is newer than the current binary notify, else do nothing.